client.go

```diff
@@ -520,6 +526,269 @@ func (c *Client) ListImages(ctx context.Context, filters ...string) ([]Image, er
 	return images, nil
 }
 
+// Checkpoint creates a checkpoint of the container
+func (c *Client) Checkpoint(ctx context.Context, id, ref string, opts ...CheckpointOpts) error {
```

Comment:
Can we provide the container instead of ID? And if we already need a container to checkpoint, why not place the checkpoint method on the container?

Reply:
It felt weird having `Checkpoint` on the container and `Restore` on the client but it doesn't matter to me.  I guess it makes sense since it's performing on the container.

Reply:
i think its fine.  Restore is just like NewContainer